### PR TITLE
Remove single view assumption from pointer events

### PR DESCRIPTION
**Pull request by loic-sharma**
This change removes the assumption that that all pointer events go to the view ID `0`. This change also adds a test for pointer events.

Part of: https://github.com/flutter/flutter/issues/112205

## Background

All pointer events are sent to [`PlatformDispatcher.onPointerDataPacket`](https://api.flutter.dev/flutter/dart-ui/PlatformDispatcher/onPointerDataPacket.html). In the future, a `viewId` property will be added [`PointerData`](https://api.flutter.dev/flutter/dart-ui/PointerData-class.html), which the framework (specifically, `GestureBinding`) will use to route the event to the proper view. This `viewId` property's value will be set by the embedders.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: runtime/runtime_controller.cc [hunk 1]
```diff
@@ -272,7 +272,7 @@ bool RuntimeController::DispatchPointerDataPacket(
     const PointerDataPacket& packet) {
   if (auto* platform_configuration = GetPlatformConfigurationIfAvailable()) {
     TRACE_EVENT0("flutter", "RuntimeController::DispatchPointerDataPacket");
-    platform_configuration->get_window(0)->DispatchPointerDataPacket(packet);
```
Comment (gspencergoog) [inline runtime/runtime_controller.cc:275]: On the framework side, how does it know which window to associate the pointer data with?  Or does it just not do that, and pointer data is just in global screen coordinates?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

This looks good to me, I just had a question for my own edification.

File: runtime/runtime_controller.cc (hunk 1 above)
Comment (loic-sharma) [reply runtime/runtime_controller.cc:275]: We'll be adding a `viewId` member to `PointerData` that will be set by the embedder. However, that requires changing the embedder API which we'll do only after we've reviewed the multi-view engine spec.

**Review by dkwingsmt — APPROVED** [review]
LGTM

**Comment by dkwingsmt** [comment]
It seems like the plan is to move all messaging methods out of `Window`, making it a data class?

**Comment by loic-sharma** [comment]
> It seems like the plan is to move all messaging methods out of Window, making it a data class?

Yup, the only other method - set viewport metrics - will also get lifted up to `PlatformConfiguration`.

**Review by cbracken — APPROVED** [review]
This looks great -- I've got nothing! Nice!

**Pull request merged**